benches/src/bin/collect.rs

```diff
@@ -570,9 +581,10 @@ impl State {
         let relative = groups.into_iter().filter_map(|(name, samples)| {
             let relative = samples
                 .into_iter()
-                .find(|sample| sample.ends_with(&name))
+                .find(|sample| sample.starts_with(&name) && ids.contains_key(sample))
```

Comment:
This fixes the name check issue.

`ids.contains_key(sample)` is also needed because the list of samples also includes any benches found previously in the `reports` folder. If the `"group_complete" line lists a bench for this group not found in a `"benchmark-complete"` line above it, the mapping from the sample to the mean would silently fail, causing the benchmark not to appear in the final output.